File: Software Development Module/Week2/MakingChange.java
/*
MakingChange.java
Billy Cussen
07/07/2020
*/

import java.util.Scanner;

public class MakingChange {
    
    public static void main(String [] args){

        String retry;

        do{

            Scanner keyboard = new Scanner(System.in);
            int fiftyEuro = 0, twentyEuro = 0, tenEuro = 0, fiveEuro = 0, twoEuro = 0, oneEuro = 0, fiftyCent = 0, twentyCent = 0, tenCent = 0, fiveCent = 0, twoCent = 0, oneCent = 0;
            double change = 0, changeTemp = 0;
            boolean checkAmount = false;
            retry = new String();

            do{
                System.out.print("Enter the cost of the Transaction: ");
                double cost = keyboard.nextDouble();
                
                System.out.print("Enter how much the customer has presented: ");
                double customer = keyboard.nextDouble();

                if(customer >= cost){
                    checkAmount = true;  
                    change = (double) Math.round((customer-cost) * 100.0) / 100.0;
                    changeTemp = change;
                }else{
                    System.out.println("The customer has not provided enough money, please ask again");
                }
            }while(!checkAmount);

            System.out.println("The customers total change is: "+change);

            while(changeTemp >= 50){
                changeTemp -= 50;
                fiftyEuro++;
            }

            if(fiftyEuro>0){
                System.out.println("50 notes in change: "+fiftyEuro);
            }

            while(changeTemp >= 20){
                changeTemp -= 20;
                twentyEuro++;
            }

            if(twentyEuro>0){
                System.out.println("20 notes in change: "+twentyEuro);
            }

            while(changeTemp >= 10){
                changeTemp -= 10;
                tenEuro++;
            }

            if(tenEuro>0){
                System.out.println("10 notes in change: "+tenEuro);
            }

            while(changeTemp >= 5){
                changeTemp -= 5;
                fiveEuro++;
            }

            if(fiveEuro>0){
                System.out.println("5 notes in change: "+fiveEuro);
            }

            while(changeTemp >= 2){
                changeTemp -= 2;
                twoEuro++;
            }

            if(twoEuro>0){
                System.out.println("2 euro coins in change: "+twoEuro);
            }

            while(changeTemp >= 1){
                changeTemp -= 1;
                oneEuro++;
            }

            if(oneEuro>0){
                System.out.println("1 euro coins in change: "+oneEuro);
            }

            while(changeTemp >= .5){
                changeTemp -= .5;
                fiftyCent++;
            }

            if(fiftyCent>0){
                System.out.println("50 cent coins in change: "+fiftyCent);
            }

            while(changeTemp >= .2){
                changeTemp -= .2;
                twentyCent++;
            }

            if(twentyCent>0){
                System.out.println("20 cent coins in change: "+twentyCent);
            }

            while(changeTemp >= .1){
                changeTemp -= .1;
                tenCent++;
            }

            if(tenCent>0){
                System.out.println("10 cent coins in change: "+tenCent);
            }

            while(changeTemp >= .05){
                changeTemp -= .05;
                fiveCent++;
            }

            if(fiveCent>0){
                System.out.println("5 cent coins in change: "+fiveCent);
            }

            while(changeTemp >= .02){
                changeTemp -= .02;
                twoCent++;
            }

            if(twoCent>0){
                System.out.println("2 cent coins in change: "+twoCent);
            }

            while(changeTemp >= .01){
                changeTemp -= .01;
                oneCent++;
            }

            if(oneCent>0){
                System.out.println("1 cent coins in change: "+oneCent);
            }
            

            while(!(retry.equalsIgnoreCase("Y")||retry.equalsIgnoreCase("N"))){

                System.out.print("Process another Transaction (Y/N): " );
                retry = keyboard.next();

            }

            keyboard = null;

        }while(retry.equalsIgnoreCase("Y"));
    }

}

File: Python Course/Software Development Module - in Python/Week2/MakingChange.py
""""
MakingChange.py
Billy Cussen
23/09/2020
"""

inputCheck = True
repeat = "y"

while repeat == "y":

    fiftyEuro = 0
    twentyEuro = 0
    tenEuro = 0
    fiveEuro = 0
    twoEuro = 0
    oneEuro = 0
    fiftyCent = 0
    twentyCent = 0
    tenCent = 0
    fiveCent = 0
    twoCent = 0
    oneCent = 0

    while inputCheck:
        try:
            cost = float(input("Enter the Cost of the Transaction: "))
        except:
            print("You need to enter a Number!")
        else:
            if cost <= 0:
                print("You need to enter a Number greater than 0")
            else:
                inputCheck = False
    
    while inputCheck == False:
        try:
            customer = float(input("Enter How much the customer has presented: "))
        except:
            print("You need to enter a Number!")
        else:
            if customer <= 0:
                print("You need to enter a Number greater than 0")
            elif customer < cost:
                print("The Customer has not provided enough, please ask again")
            else:
                inputCheck = True

    change = float(round(((customer-cost)*100),2)/100)
    print("Customer Change is: "+str(change))

    while change >= 50:
        change-=50
        fiftyEuro = fiftyEuro + 1
    if fiftyEuro > 0:
        print("Fifty Euro Notes in Change: "+str(fiftyEuro))

    while change >= 20:
        change-=20
        twentyEuro = twentyEuro + 1
    if twentyEuro > 0:
        print("Twenty Euro Notes in Change: "+str(twentyEuro))

    while change >= 10:
        print(str(change))
        change-=10
        tenEuro = tenEuro + 1
    if tenEuro > 0:
        print("Ten Euro Notes in Change: "+str(tenEuro))

    while change >= 5:
        change-=5
        fiveEuro = fiveEuro + 1
    if fiveEuro > 0:
        print("Five Euro Notes in Change: "+str(fiveEuro))  

    while change >= 2:
        change-=2
        twoEuro = twoEuro + 1
    if twoEuro > 0:
        print("Two Euro Coins in Change: "+str(twoEuro))  

    while change >= 1:
        change-=1
        oneEuro = oneEuro + 1
    if oneEuro > 0:
        print("One Euro Coins in Change: "+str(oneEuro))  

    while change >= .5:
        change-=.5
        fiftyCent = fiftyCent + 1
    if fiftyCent > 0:
        print("Fifty Cent Coins in Change: "+str(fiftyCent))  

    while change >= .2:
        change-=.2
        twentyCent = twentyCent + 1
    if twentyCent > 0:
        print("Twenty Cent Coins in Change: "+str(twentyCent)) 

    repeat = ""

    while repeat != "y" and repeat != "n":
        repeat = str(input("Would you like to try again (\"Y\" or \"N\"): "))
        repeat = repeat.lower()
        if(repeat != "y" and repeat != "n"):
            print("You need to enter Y or N")
    
print("Thanks for using the App!")
